census_data_downloader/base.py

```diff
@@ -234,27 +278,41 @@ def download_usa(self):
         """
         Download all datasets that provide full coverage for the entire country.
         """
-        self.download_nationwide()
-        self.download_states()
-        self.download_congressional_districts()
-        self.download_counties()
-        self.download_places()
-        self.download_urban_areas()
-        self.download_msas()
-        self.download_csas()
-        self.download_pumas()
-        self.download_aiann_homelands()
-        self.download_zctas()
+        if "nationwide" in self.GEO_LIST:
+            self.download_nationwide()
+        if "state" in self.GEO_LIST:
+            self.download_states()
+        if "congressional_district" in self.GEO_LIST:
+            self.download_congressional_districts()
+        if "county" in self.GEO_LIST:
+            self.download_counties()
+        if "place" in self.GEO_LIST:
+            self.download_places()
+        if "urban_area" in self.GEO_LIST:
+            self.download_urban_areas()
+        if "msa" in self.GEO_LIST:
+            self.download_msas()
+        if "csa" in self.GEO_LIST:
+            self.download_csas()
+        if "puma" in self.GEO_LIST:
+            self.download_pumas()
+        if "aiannh_homeland" in self.GEO_LIST:
+            self.download_aiann_homelands()
+        if "zcta" in self.GEO_LIST:
+            self.download_zctas()
```

Comment:
Could we simplify this a little by looping through the GEO_LIST and pulling the methods dynamically one by one. Here's some rough psuedo code that I think approximates what might work.

```python
# Loop through the geographies
for geo in self.GEO_LIST:
   # Get each download method
   func = getattr(self, f"download_{geo}")
   # Call each download method
   func()
```

This will requires that slightly variations like "msa" versus "download_msas" be sorted out. But I bet can do that by simply changing the existing download method names to match the new geolist.